packages/image-block/src/ImageBlock.tsx

```diff
@@ -1,21 +1,72 @@
 /* (c) Copyright Frontify Ltd., all rights reserved. */
 
-import { useBlockAssets, useBlockSettings, useEditorState } from '@frontify/app-bridge';
+import { Asset, useBlockAssets, useBlockSettings, useEditorState } from '@frontify/app-bridge';
 import '@frontify/fondue-tokens/styles';
 import { BlockProps } from '@frontify/guideline-blocks-settings';
 import 'tailwindcss/tailwind.css';
 import { Settings, mapCaptionPositionClasses } from './types';
 import { UploadPlaceholder } from './components/UploadPlaceholder';
 import { ImageCaption } from './components/ImageCaption';
-import { IMAGE_SETTING_ID } from './settings';
+import { ATTACHMENTS_SETTING_ID, IMAGE_SETTING_ID } from './settings';
 import { joinClassNames } from '@frontify/guideline-blocks-shared';
 import { Image } from './components/Image';
 
 export const ImageBlock = ({ appBridge }: BlockProps) => {
     const [blockSettings, setBlockSettings] = useBlockSettings<Settings>(appBridge);
-    const { blockAssets } = useBlockAssets(appBridge);
+    const { blockAssets, updateAssetIdsFromKey, addAssetIdsToKey, deleteAssetIdsFromKey } = useBlockAssets(appBridge);
     const image = blockAssets?.[IMAGE_SETTING_ID]?.[0];
+    const attachments = blockAssets?.[ATTACHMENTS_SETTING_ID] || [];
     const isEditing = useEditorState(appBridge);
+    const attachmentsInBlockSettings = blockSettings.attachments || [];
+
+    const onAddAttachments = (assets: Asset[]) => {
```

Comment:
you could use `useReducer` to extract this logic and pass down the dispatch action:
```
const reducer = (state, action) => {
    switch (action.type) {
        case 'add':
            ...
        case 'replace':
            ...
        case 'delete':
            ...
        case 'sorted':
            ...
}
```

see https://beta.reactjs.org/reference/react/useReducer

Reply:
that way, you can passt down the dispatch action which you get from `useReducer` to avoid prop drilling

Reply:
regarding implementation: for the settings itself, it would mean to introduce useReducedSettings in appBridge

Reply:
Thanks Marco for the insight. I decided to create a useAttachments hook inside the shared folder with all the logic bundled in it, so it is easily reusable. Let me know what do you think!